torch/_dynamo/output_graph.py

```diff
@@ -552,6 +552,20 @@ def real_value_cache(self):
     def bound_symbols(self):
         return self.current_tracer.bound_symbols
 
+    # Unbacked symbols may be created as a function call in parent (e.g. t.to_list()).
+    # We track the symbol-proxy mapping for unbacked symbols when the
+    # SymNodeVariable is constructed (e.g. t.to_list() call) so that when subtracer tries
+    # to bind an unbacked symint, e.g. a subgraph input with shape u0,
+    # the proxy for u0 can be found in parent's bound_symbols.
+    def track_unbacked_symbols(self, s: torch.SymInt, proxy: torch.fx.Proxy):
```

Comment:
Is it guaranteed that the symint isn't tracked in the current tracer (assuming nested tracers)? If so, can we add some asserts to that effect

Reply:
> def f(x):
  def true_f():
    c =  x.item()
    def true_f_inner():
      return c
      When x.item() is called, proxy_of_c and its symnode variable is created for tracer 1, and we also call track_unbacked_symbols to record this relationship. So when tracer 2 finds proxy_of_c is not created by current tracer, it recursivelly looks up its parent tracer and finds that that expression u0 has been tracked as a result of track_unbacked_symbol in tracer 1. So it will stop the recursion and create_graph_input u0 in tracer 2. Graph looks like:
      
 Based on this description, I would expect the `proxy` always be from the current tracer, is that correct? (if so, add asserts)